src/data/set/countable.lean

```diff
@@ -42,71 +50,67 @@ set.countable_iff_exists_injective.trans
      hf $ by simpa [as, bs] using h⟩,
  λ ⟨f, hf⟩, ⟨_, inj_on_iff_injective.1 hf⟩⟩
 
-lemma countable_iff_exists_subset_range [ne : nonempty α] {s : set α} :
-  s.countable ↔ ∃f:ℕ → α, s ⊆ range f :=
-⟨λ ⟨h⟩, by inhabit α; exactI ⟨λ n, ((decode s n).map subtype.val).iget,
-  λ a as, ⟨encode (⟨a, as⟩ : s), by simp [encodek]⟩⟩,
- λ ⟨f, hf⟩, ⟨⟨
-  λ x, inv_fun f x.1,
-  λ n, if h : f n ∈ s then some ⟨f n, h⟩ else none,
-  λ ⟨x, hx⟩, begin
-    have := inv_fun_eq (hf hx), dsimp at this ⊢,
-    simp [this, hx]
-  end⟩⟩⟩
+/-- Convert `set.countable s` to `encodable s` (noncomputable). -/
+protected def countable.to_encodable {s : set α} : s.countable → encodable s :=
+classical.choice
+
+section enumerate
+
+/-- Enumerate elements in a countable set.-/
```

Comment:
While we're refactoring, maybe something like
```suggestion
/-- Noncomputably enumerate elements in a set. The `default` value is used to extend the domain to all of `ℕ`. -/
```
If you have the time, it would be nice to change the definition to use `option.option.get_or_else`. (It might be even nicer to have this take a proof that `s` is nonempty so there are better lemmas about it, but maybe not this PR.)